From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stefan Hertrampf <stefan.hertrampf@cyberus-technology.de>
Date: Mon, 5 Feb 2024 13:13:50 +0100
Subject: [PATCH] config: switch to split irq chip as default

---
 Config.kmk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Config.kmk b/Config.kmk
index d353b0920..7d09ca397 100644
--- a/Config.kmk
+++ b/Config.kmk
@@ -2009,7 +2009,6 @@ endif
 CYBERUS_CXX_FLAGS = -Werror -Wall
 ifdef VBOX_WITH_KVM
  DEFS += VBOX_WITH_KVM
- DEFS += VBOX_WITH_KVM_IRQCHIP_FULL
 endif
 ifndef VBOX_HEADLESS
  DEFS += VBOX_WITH_GVT_RENDERING
-- 
2.44.0

